File: web/java/huntkingdom/src/Entities/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import java.util.Date;

/**
 *
 * @author tibh
 */
public class Produit {
    private int id;
    private String lib_prod;
    private Double prix;
    private String description;
    private int qte_prod;
     private Date date_ajout;
    private String image;
    private String type;
    
    
    public Produit(){}
    
    public Produit(int id, String lib_prod, Double prix, String description, int qte_prod, Date date_ajout, String image, String type) {
        this.id = id;
        this.lib_prod = lib_prod;
        this.prix = prix;
        this.description = description;
        this.qte_prod = qte_prod;
        this.date_ajout = date_ajout;
        this.image = image;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLib_prod() {
        return lib_prod;
    }

    public void setLib_prod(String lib_prod) {
        this.lib_prod = lib_prod;
    }

    public Double getPrix() {
        return prix;
    }

    public void setPrix(Double prix) {
        this.prix = prix;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getQte_prod() {
        return qte_prod;
    }

    public void setQte_prod(int qte_prod) {
        this.qte_prod = qte_prod;
    }

    public Date getDate_ajout() {
        return date_ajout;
    }

    public void setDate_ajout(Date date_ajout) {
        this.date_ajout = date_ajout;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id + ", lib_prod=" + lib_prod + ", prix=" + prix + ", description=" + description + ", qte_prod=" + qte_prod + ", date_ajout=" + date_ajout + ", image=" + image + ", type=" + type + '}';
    }
   
}

File: src/ProductBundle/Entity/Produit.php
<?php

namespace ProductBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Produit
 *
 * @ORM\Table(name="produit")
 * @ORM\Entity(repositoryClass="ProductBundle\Repository\ProduitRepository")
 */
class Produit
{
    /**
     * Many Users have Many Groups.
     * @ORM\ManyToMany(targetEntity="Panier", inversedBy="produits")
     * @ORM\JoinTable(name="commandes")
     */
    private $paniers;

    public function __construct() {
        $this->paniers = new \Doctrine\Common\Collections\ArrayCollection();
    }
    /**
     * @ORM\ManyToOne(targetEntity="Promotion")
     * @ORM\JoinColumn(name="promotion_id", referencedColumnName="id")
     */
    private $promotion;



    /**
     * @return \Doctrine\Common\Collections\ArrayCollection
     */
    public function getPaniers()
    {
        return $this->paniers;
    }

    /**
     * @param \Doctrine\Common\Collections\ArrayCollection $paniers
     */
    public function setPaniers($paniers)
    {
        $this->paniers = $paniers;
    }

    /**
     * @return mixed
     */
    public function getPromotion()
    {
        return $this->promotion;
    }

    /**
     * @param mixed $promotion
     */
    public function setPromotion($promotion)
    {
        $this->promotion = $promotion;
    }



    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="lib_prod", type="string", length=255)
     */
    private $libProd;

    /**
     * @var float
     *
     * @ORM\Column(name="prix", type="float")
     */
    private $prix;

    /**
     * @var float
     *
     * @ORM\Column(name="prixFinale", type="float")
     */
    private $prixFinale;

    /**
     * @return float
     */
    public function getPrixFinale()
    {
        return $this->prixFinale;
    }

    /**
     * @param float $prixFinale
     */
    public function setPrixFinale($prixFinale)
    {
        $this->prixFinale = $prixFinale;
    }

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="text")
     */
    private $description;

    /**
     * @var int
     *
     * @ORM\Column(name="qte_prod", type="integer")
     */
    private $qteProd;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_ajout", type="date")
     */
    private $dateAjout;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=255)
     */
    private $image;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=255)
     */
    private $type;

    /**
     * @var string
     *
     * @ORM\Column(name="marque", type="string", length=255)
     */
    private $marque;

    /**
     * @return string
     */
    public function getMarque()
    {
        return $this->marque;
    }

    /**
     * @param string $marque
     */
    public function setMarque($marque)
    {
        $this->marque = $marque;
    }

    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set libProd
     *
     * @param string $libProd
     *
     * @return Product
     */
    public function setLibProd($libProd)
    {
        $this->libProd = $libProd;

        return $this;
    }

    /**
     * Get libProd
     *
     * @return string
     */
    public function getLibProd()
    {
        return $this->libProd;
    }

    /**
     * Set prix
     *
     * @param float $prix
     *
     * @return Product
     */
    public function setPrix($prix)
    {
        $this->prix = $prix;

        return $this;
    }

    /**
     * Get prix
     *
     * @return float
     */
    public function getPrix()
    {
        return $this->prix;
    }

    /**
     * Set description
     *
     * @param string $description
     *
     * @return Product
     */
    public function setDescription($description)
    {
        $this->description = $description;

        return $this;
    }

    /**
     * Get description
     *
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * Set qteProd
     *
     * @param integer $qteProd
     *
     * @return Product
     */
    public function setQteProd($qteProd)
    {
        $this->qteProd = $qteProd;

        return $this;
    }

    /**
     * Get qteProd
     *
     * @return int
     */
    public function getQteProd()
    {
        return $this->qteProd;
    }

    /**
     * Set dateAjout
     *
     * @param \DateTime $dateAjout
     *
     * @return Product
     */
    public function setDateAjout($dateAjout)
    {
        $this->dateAjout = $dateAjout;

        return $this;
    }

    /**
     * Get dateAjout
     *
     * @return \DateTime
     */
    public function getDateAjout()
    {
        return $this->dateAjout;
    }

    /**
     * Set image
     *
     * @param string $image
     *
     * @return Product
     */
    public function setImage($image)
    {
        $this->image = $image;

        return $this;
    }

    /**
     * Get image
     *
     * @return string
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * Set type
     *
     * @param string $type
     *
     * @return Product
     */
    public function setType($type)
    {
        $this->type = $type;

        return $this;
    }

    /**
     * Get type
     *
     * @return string
     */
    public function getType()
    {
        return $this->type;
    }
}

